Source/core/SocketPort.cpp

```diff
@@ -238,7 +239,11 @@ namespace Thunder {
                         interfaceId = info->ipi_ifindex;
                         break;
                     }
+#ifdef __APPLE__
```

Comment:
To avoid a lot of single line preprocessor descriminitors, look at the top of this file and define the actual type for each platform, so for example at the beginning of this file do, for each platform:
#define __PACKAGE_TYPE__ IPV6_RECVPKTINFO
and
#define __PACKAGE_TYPE__ IPV6_PKTINFO
and than here, use the __PACKAGE_TYPE__ it will keep the code easier to read..

Reply:
done